main.tf

```diff
@@ -271,7 +271,7 @@ DOC
 
 resource "aws_launch_template" "default" {
   name_prefix   = module.this.id
-  image_id      = length(var.ami) > 0 ? var.ami : data.aws_ami.amazon_linux_2.id
+  image_id      = length(var.ami) > 0 ? var.ami : data.aws_ami.amazon_linux_2023.id
```

Comment:
We can do this cleaner with `coalesce()`

https://developer.hashicorp.com/terraform/language/functions/coalesce

Reply:
Aha, thanks for that